src/offchainapi/protocol_messages.py

```diff
@@ -265,53 +285,60 @@ def make_success_response(request):
     return response
 
 
-def make_protocol_error(request, code=None):
+def make_protocol_error(request, code, message=None):
     """ Constructs a CommandResponse signaling a protocol failure.
         We do not sequence or store such responses since we can recover
         from them.
 
     Args:
         request (CommandRequestObject): The request object.
-        code (int or None, optional): The error code. Defaults to None.
+        code (OffchainErrorCode): The error code.
+        message (str, optional): A human-readable message about this error.
 
     Returns:
         CommandResponseObject: The generated response object.
     """
     response = CommandResponseObject()
     response.cid = request.cid
     response.status = 'failure'
-    response.error = OffChainError(protocol_error=True, code=code)
+    response.error = OffChainError(protocol_error=True, code=code, message=message)
```

Comment:
I tried to clarify in the comment:

    class OffChainProtocolError(Exception):
    ''' This class denotes protocol errors, namely errors at the
        OffChain protocols level rather than the command sequencing level.

        This is an Exception that is thown within the Python program
        to represent the error, rather than the message type which is
        OffChainError.
        '''